### ISSUE TITLE
div get dimensions

### ISSUE THREAD

**Issue by shiffman** [processing/p5.js#391]
A small thing, but also re: p5.dom a lot of my examples require dynamically accessing the size of a DIV (which changes according to the amount of text).  I may be mistaken but I believe p5 only allows for setting the dimensions with `size()` but for retrieving the width/height I am doing the following:

```
var w = div.elt.offsetWidth;
var h = div.elt.offsetHeight;
```

Would it make sense to have a `width()/height()` method that returns the dimensions?

**Comment by futuremarc** [comment]
.size() now returns a string with the current width & height of the element if no arguments are provided. 

However, p5 Elements already have the variables 'width' and 'height' that store the element's initial offsetWidth/offsetHeight. Should we use a naming convention like eltWidth and eltHeight?

**Comment by shiffman** [comment]
I ran some quick tests (with the following code).

```
var txt;

function setup() {
  noCanvas();
  txt = createDiv("Hello!");
  txt.style("background-color","#00FF00");
  txt.position(100,100);
  console.log(txt.width);
  txt.html("Hello again!")
  console.log(txt.width); 
}

function mousePressed() {
  console.log(txt.width);
  console.log(txt.size());
}
```

The current behavior I believe is:

* `size()` returns the `p5.Element` instance.  I like the idea of it returning dimensions, but a string is not particularly useful as it will almost always need to be parsed back into numbers.   What if `size()` returns an object with a width and height, i.e.
```
{
  "width": elt.offsetWidth,
  "height: elt.offsetHeight
}
```
* `width` and `height` both return the original width and height of the element no matter how much it changes.  This would be a perfect solution as far as I'm concerned but is that original width/height being maintained for a reason I'm not aware of?  An alternative would be to add `width()` and `height()` functions but that seems like overkill given we have `size()` already.

**Comment by futuremarc** [comment]
I think and object returned from .size() is enough.

In this spirit of doubling as a getter/setter, what if with null arguments things like `.style()` return an object of the elt's styles.

This could also help when considering how to more easily deal with input events. We can make the setter/getter `.events()` which would either return an object of events or set an event handler.

Something like:

`var btn = createButton();
var div = createDiv('im a div');
btn.events().onmousedown = div.html('listening to btn');
`

**Comment by lmccart** [comment]
I think this syntax is a little hard to follow. We have a mousedown event handler actually.

```javascript
btn.mousePressed( function(e) { // do stuff } )
```

If we wanted to add a generic `event()` method, I think a format like `event(eventType, func)` would follow the form set by `style()`.
```javascript
btn.event('mousedown', function(e) { // do stuff } );
```

In any case, the right side of the code above would need to be wrapped inside of a function, so it's not fired immediately, something like:

```javascript
 =  function() { div.html('listening to btn'); };
```

**Comment by lmccart** [comment]
re: `width` and `height`, this sounds like a bug. they should be updated when the size changes, though there is still room for these to be wrong. for example, if you created an element and set the width and height manually with css rather than using the `style()` function, so having `size()` return the width and height makes sense.

**Comment by futuremarc** [comment]
* `width` and `height` now return the correct value and they update on change. (though only act as getters)
* `size()` returns an object of `width` and `height` if empty.
* `size()` actually sets the size of an element now (the css markup `!important` was breaking it).
* `position()` returns an object of `x` and `y`  if empty.
* `x` and `y` are now attributes on the element and can be accessed via `elt.x` or `elt.y` (to stay consistent. re: #579 )

**Comment by futuremarc** [comment]
closed with [9015ffe](https://github.com/processing/p5.js/commit/9015ffe3eab19c155e42a1dc7be381974fc689be)
